### DISCUSSION TITLE
JWT payload format for use with Saleor as a Resource server

### DISCUSSION THREAD

**Discussion by looped-monk** [saleor/saleor#10558] [General]
Hello ,

I've created the payload endpoints for JWKS and User info on my Django blogging app for use with Saleor as a resource server. I need to know if this is the right format. The same endpoints logically will be used by front-end app and tokens will passed to Saleor-Docker.

User_info endpoint will return this : 

`def _ecommerce_verification_token_for_jwt(self):   
        return jwt.encode({
        "token_type": "Bearer",
        "access_token": "my_access_token",
        "user_id": self.pk,
        "client_id":settings.APPLICATION_ID_ECOMMERCE,
        "client_secret":settings.SECRET_KEY,
        "refresh":"refresh",
        "id":self.Post.uuid,
        "name":self.Post.title,
        "email": self.email,
        "exp": datetime.utcnow() + timedelta(days=1)},
         algorithm=settings.JWT_ALGORITHM_ECOMMERCE).decode('utf-8')
`

JWKS-endpoint will return this : 

`def _json_web_key_set(self):
        return jwt.encode({
                "keys": [{
                "kty": "RSA",
                "e": "my_value",
                "kid": "my_value",
                "n": "my_value",
                "exp": datetime.utcnow() + timedelta(days=1)}]},      
                algorithm=settings.JWT_ALGORITHM_ECOMMERCE).decode('utf-8')
`

**Comment by patrys** [comment]
Are you trying to use your blog as an external auth server?

**Reply by looped-monk** [reply]
@patrys Yup. I am.

**Reply by looped-monk** [reply]
@patrys Any update on above ? I intend to send the above payload to Saleor for authentication. Im looking at Saleor as a marketplace model. I couldn't find any feature pertaining to the same where users can be both vendors and customers themselves and Saleor is merely for order fulfillment.

**Reply by patrys** [reply]
I would suggest that you take a look at the OIDC plugin. I think it would be easier for you to expose an established authentication/authorization protocol (OpenID Connect) endpoint on your blog end, which would make Saleor work out of the box.
